### PR TITLE
Support [DefaultValue] when new feature switch is disabled

### LINKED ISSUE #103542: Allow `DefaultValueAttribute` with primitives when `System.ComponentModel.DefaultValueAttribute.IsSupported` disabled
### Description

`DefaultValueAttribute.Value` throws on access (due to https://github.com/dotnet/runtime/pull/100416) when `System.ComponentModel.DefaultValueAttribute.IsSupported` switch is `false` and it breaks `XmlSerializer` and other serializers/reflection. But it's very common to use not `(Type, string)` overload, but 1 arg primitive which do not use `TypeDescriptor` API family.

### Reproduction Steps

```cs
using System.ComponentModel;
using System.Xml.Serialization;

// _ = typeof(C).GetProperty("i")!.GetCustomAttribute<DefaultValueAttribute>()!.Value;
XmlSerializer serializer = new(typeof(C));

public class C { [DefaultValue(5)] public int i {get; set;} }
```
```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net9.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <RuntimeHostConfigurationOption
      Include="System.ComponentModel.DefaultValueAttribute.IsSupported"
      Value="false" Trim="true" />
  </ItemGroup>

</Project>
```

### Expected behavior

No exception, use constructor arg as is.

### Actual behavior

```
Unhandled exception. System.InvalidOperationException: There was an error reflecting type 'C'.
 ---> System.ArgumentException: Runtime instantiation of this attribute is not allowed.
   at System.ComponentModel.DefaultValueAttribute.get_Value()
   at System.Xml.Serialization.XmlAttributes..ctor(ICustomAttributeProvider provider)
   at System.Xml.Serialization.XmlReflectionImporter.GetAttributes(MemberInfo memberInfo)
   at System.Xml.Serialization.XmlReflectionImporter.InitializeStructMembers(StructMapping mapping, StructModel model, Boolean openModel, String typeName, RecursionLimiter limiter)
   at System.Xml.Serialization.XmlReflectionImporter.ImportStructLikeMapping(StructModel model, String ns, Boolean openModel, XmlAttributes a, RecursionLimiter limiter)
   at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping(TypeModel model, String ns, ImportContext context, String dataType, XmlAttributes a, Boolean repeats, Boolean openModel, RecursionLimiter limiter)
   --- End of inner exception stack trace ---
   at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping(TypeModel model, String ns, ImportContext context, String dataType, XmlAttributes a, Boolean repeats, Boolean openModel, RecursionLimiter limiter)
   at System.Xml.Serialization.XmlReflectionImporter.ImportElement(TypeModel model, XmlRootAttribute root, String defaultNamespace, RecursionLimiter limiter)
   at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping(Type type, XmlRootAttribute root, String defaultNamespace)
   at System.Xml.Serialization.XmlSerializer..ctor(Type type, String defaultNamespace)
   at Program.<Main>$(String[] args) in C:\Source\ttt\Program.cs:line 4
```

### Workaround

set `_value` to some sentinel if `TypeDescriptor` is required and throw only in this case.

### PR DESCRIPTION
**Pull request by steveharter**
Fixes https://github.com/dotnet/runtime/issues/103542

Throw only on the (Type, String) ctor.

### CODE REVIEW

**Review by LakshanF — APPROVED** [review]
LGTM!

File: src/libraries/System.Private.CoreLib/src/System/ComponentModel/DefaultValueAttribute.cs [hunk 1]
```diff
@@ -245,21 +247,22 @@ public DefaultValueAttribute(ulong value)
         {
             get
             {
-                if (!IsSupported)
+                if (!IsSupported && object.ReferenceEquals(_value, s_throwSentinel))
```
Comment (teo-tsirpanis) [inline src/libraries/System.Private.CoreLib/src/System/ComponentModel/DefaultValueAttribute.cs:250]:
```suggestion
                if (!IsSupported && ReferenceEquals(_value, s_throwSentinel))
```

**Comment by steveharter** [comment]
/ba-g Created new Quic issue for unrelated failure

**Comment by akoeplinger** [comment]
This caused a test failure on dependency flow to sdk, please take a look: https://github.com/dotnet/sdk/pull/42496#issuecomment-2263663666

**Pull request merged**